test/sequential/bootstrap.updatePromotion.test.ts

```diff
@@ -0,0 +1,214 @@
+/* eslint-disable jest/no-conditional-expect */
+/*
+ *
+ * @group bootstrap
+ * @group sequential
+ */
+import { getApi, initApi } from "../../utils/api";
+import {
+  scheduleBootstrap,
+  finalizeBootstrap,
+  updatePromoteBootstrapPool,
+  provisionBootstrap,
+  claimRewardsBootstrap,
+  getLiquidityAssetId,
+  getBalanceOfAsset,
+} from "../../utils/tx";
+import { EventResult, ExtrinsicResult } from "../../utils/eventListeners";
+import { Keyring } from "@polkadot/api";
+import { User } from "../../utils/User";
+import {
+  getEnvironmentRequiredVars,
+  waitForBootstrapStatus,
+} from "../../utils/utils";
+import {
+  getEventResultFromMangataTx,
+  getEventErrorfromSudo,
+} from "../../utils/txHandler";
+import {
+  checkLastBootstrapFinalized,
+  createNewBootstrapCurrency,
+  setupBootstrapTokensBalance,
+} from "../../utils/Bootstrap";
+import { MGA_ASSET_ID } from "../../utils/Constants";
+import { BN, MangataGenericEvent } from "@mangata-finance/sdk";
+import { setupUsers } from "../../utils/setup";
+
+jest.spyOn(console, "log").mockImplementation(jest.fn());
+jest.spyOn(console, "error").mockImplementation(jest.fn());
+jest.setTimeout(3500000);
+process.env.NODE_ENV = "test";
+
+let testUser1: User;
+let sudo: User;
+let keyring: Keyring;
+let bootstrapCurrency: any;
+let bootstrapPool: any;
+let eventResponse: EventResult;
+
+const { sudo: sudoUserName } = getEnvironmentRequiredVars();
+//constant for bootstrap include a planning period
+const waitingPeriodWithPlan = 400;
+//constant for bootstrap less a planning period
+const waitingPeriodLessPlan = 10;
+const bootstrapPeriod = 20;
+const whitelistPeriod = 10;
+const bootstrapAmount = new BN(10000000000);
+
+async function changePromoteBootstrapPool(userName: User) {
+  const api = getApi();
+
+  const currentPromotingState =
+    await api.query.bootstrap.promoteBootstrapPool();
+
+  const result = await updatePromoteBootstrapPool(
+    userName,
+    !currentPromotingState
+  );
+
+  return result;
+}
+
+async function checkSudoOperataionSuccess(
+  checkingEvent: MangataGenericEvent[]
+) {
+  const filterBootstrapEvent = checkingEvent.filter(
+    (extrinsicResult) => extrinsicResult.method === "Sudid"
+  );
+
+  const userBootstrapCall = filterBootstrapEvent[0].event.data[0].toString();
+
+  expect(userBootstrapCall).toContain("Ok");
+}
+
+async function checkSudoOperataionFail(
+  checkingEvent: MangataGenericEvent[],
+  expectedError: string
+) {
+  const filterBootstrapEvent = checkingEvent.filter(
+    (extrinsicResult) => extrinsicResult.method === "Sudid"
+  );
+
+  const BootstrapError = await getEventErrorfromSudo(filterBootstrapEvent);
+
+  expect(BootstrapError.method).toContain(expectedError);
+}
```

Comment:
are not `checkSudoOperataionFail`  and `checkSudoOperataionSuccess` duplicated? perhaps you can extract it to a commonplace.

Reply:
They are similar, and I can use this as one function, or maybe can  I add them into, for example, `eventListeners.ts`? But in any case, there is a difference between them (in `checkSudoOperataionSuccess `result is always "Ok", and in `checkSudoOperataionFail `result changes)

Reply:
What I mean is that these functions does exist in 
test/sequential/bootstrap.updatePromotion.test.ts
And
test/sequential/bootstrap.cancelBootstrap.test.ts

Reply:
and they look the same to me

Reply:
Added as function. More in DMs